+sw_tests/+unit_tests/unittest_spinw_addg.m

```diff
@@ -0,0 +1,60 @@
+classdef unittest_spinw_addg < matlab.mock.TestCase
+
+    properties
+        swobj = [];
+    end
+
+    methods (TestMethodSetup)
+        function setup_spinw_model(testCase)
+            testCase.swobj = spinw();
+            testCase.swobj.genlattice('spgr','I 4'); % body-centred
+            testCase.swobj.addatom('r',[0,0,0], 'S',1)
+            testCase.swobj.addmatrix('label','g1','value',diag([2 1 1]))
```

Comment:
I think that the g tensor must be positive definite (its eigenvalues must be positive real values) or maybe positive semi-definite (eigenvalues can be zero)... not sure how to enforce this without calculating the eigenvalues...

[We can't check it here in `addmatrix` but maybe in `addg`?]

Reply:
Agreed we don't know here what type of interaction a matrix corresponds to. but it would be good to validate any conditions we know hold true at the `addg` stage - I think calculating the eigenvalues is not going to be too costly to do.